lib/environment.js

```diff
@@ -8,94 +8,35 @@ var ENV = function(opts) {
   this.rootPath = opts.rootPath || '.';
   this.sourcePath = opts.sourcePath || 'source';
   this.ignorePrefixPattern = opts.ignorePrefixPattern || new RegExp('\/_');
-
-  this.trail = new hike.Trail(this.rootPath);
-  this.trail.paths.append(this.sourcePath);
 };
 
-ENV.prototype.getSourceFiles = function(whitelist) {
+ENV.prototype.getSourceFiles = function(whitelist, callback) {
   var _this = this;
   whitelist = whitelist || [];
 
-  return this.getSourcePaths(whitelist).map(function(sourcePath){
-    return new SourceFile({
-      name: path.basename(sourcePath),
-      path: sourcePath
-    });
-  });
-};
-
-ENV.prototype.getSourcePaths = function(whitelist) {
-  var _this = this;
-  var paths = [];
-
-  // If we are requesting only one path, let's speed things up
-  // by using the trail.find method instead of traversing the 
-  // entire tree.
-  if(whitelist.length == 1) {
-    var singlePath = this.trail.find(whitelist[0]);
-    if(!!singlePath) {
-      paths.push(path.relative(this.rootPath, singlePath));
-    }
-
-    return paths;
-  };
-
-  this.eachEntry(this.sourcePath, function(filePath) {
-    if (_this.trail.stat(filePath).isDirectory()) {
-      return;
-    }
-
-    // Should the path be ignored based on the ignorePrefix?
-    var matches = _this.ignorePrefixPattern.exec(filePath);
-    if (matches) {
-      return;
-    }
-
-    // Should the path be filtered out because it's not 
-    // in the requested paths?
-    var whitelisted = true;
+  recursiveReaddir(_this.sourcePath, ['_*'], function (err, files) {
```

Comment:
Woops, yes!
 On Feb 3, 2015 5:54 PM, "Philip von Bargen" <notifications@github.com>
wrote:

> In lib/environment.js
> <https://github.com/spotify/lingon/pull/123#discussion_r24018896>:
>
> > -
> > -  this.eachEntry(this.sourcePath, function(filePath) {
> > -    if (_this.trail.stat(filePath).isDirectory()) {
> > -      return;
> > -    }
> > -
> > -    // Should the path be ignored based on the ignorePrefix?
> > -    var matches = _this.ignorePrefixPattern.exec(filePath);
> > -    if (matches) {
> > -      return;
> > -    }
> > -
> > -    // Should the path be filtered out because it's not
> > -    // in the requested paths?
> > -    var whitelisted = true;
> > +  recursiveReaddir(_this.sourcePath, ['_*'], function (err, files) {
>
> Shouldn't this be the ignorePrefixPattern variable instead of hardcoding
> _*?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/lingon/pull/123/files#r24018896>.
>